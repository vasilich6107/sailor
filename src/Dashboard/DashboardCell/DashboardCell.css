.dashboard__cell {
    border: 1px solid black;
    border-radius: 5px;
    padding: 10px;
    margin: 15px;
    min-width: 300px;
    align-self: flex-start;
    display: flex;
    flex-direction: column;
}

.dashboard__cell-controls {
    display: flex;
    justify-content: space-between;
}

.dashboard__cell-controls button {
    font-size: 16px;
    font-weight: bold;
}

.applicant {
    display: flex;
    flex-direction: row;
    align-self: flex-start;
}

.applicant__picture IMG {
    border-radius: 50%;
    margin: 0 15px;
}