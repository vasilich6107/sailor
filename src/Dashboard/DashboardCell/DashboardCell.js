import React from 'react';

import './DashboardCell.css';

const DashboardCell = ({ applicant, onPrevious, onNext }) => (
  <div className="dashboard__cell">
    <div className="applicant">
      <div className="applicant__picture">
        <img src={applicant.picture.medium} alt={applicant.name.first} />
      </div>
      <div className="applicant__data">
        <div>First name: {applicant.name.first}</div>
        <div>Last name: {applicant.name.last}</div>
        <div>Country: {applicant.location.city}</div>
      </div>
    </div>
    <div className="dashboard__cell-controls">
      <div>
        {onPrevious && (
          <button id="previous" type="button" onClick={() => onPrevious(applicant.login.uuid)}>
            &#8249;
          </button>
        )}
      </div>

      <div>
        {onNext && (
          <button id="next" type="button" onClick={() => onNext(applicant.login.uuid)}>
            &#8250;
          </button>
        )}
      </div>
    </div>
  </div>
);

export default DashboardCell;
