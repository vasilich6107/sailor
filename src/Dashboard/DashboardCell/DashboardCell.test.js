import React from 'react';
import { shallow } from 'enzyme';
import DashboardCell from './DashboardCell';

const applicant = {
  location: {
    city: 'bath',
  },
  name: {
    first: 'francis',
    last: 'stewart',
    title: 'mr',
  },
  login: {
    uuid: '5d204000-387b-4741-8ba9-7c121ae6adab',
  },
  picture: {
    large: 'https://randomuser.me/api/portraits/men/15.jpg',
    medium: 'https://randomuser.me/api/portraits/med/men/15.jpg',
    thumbnail: 'https://randomuser.me/api/portraits/thumb/men/15.jpg',
  },
};

it('Renders dashboard cell', () => {
  expect(shallow(<DashboardCell applicant={applicant} />)).toMatchSnapshot();
});

it('Renders dashboard cell with "previous" button', () => {
  const rev = jest.fn();
  const component = shallow(<DashboardCell applicant={applicant} onPrevious={rev} />);
  expect(component).toMatchSnapshot();

  component.find('#previous').simulate('click');
  expect(rev.mock.calls.length).toBe(1);
  expect(rev.mock.calls[0][0]).toBe(applicant.login.uuid);
});

it('Renders dashboard cell with "next" button', () => {
  const ff = jest.fn();
  const component = shallow(<DashboardCell applicant={applicant} onNext={ff} />);
  expect(component).toMatchSnapshot();

  component.find('#next').simulate('click');
  expect(ff.mock.calls.length).toBe(1);
  expect(ff.mock.calls[0][0]).toBe(applicant.login.uuid);
});

it('Renders dashboard cell with both buttons', () => {
  const rev = jest.fn();
  const ff = jest.fn();
  expect(shallow(<DashboardCell applicant={applicant} onNext={ff} onPrevious={rev} />)).toMatchSnapshot();
});
