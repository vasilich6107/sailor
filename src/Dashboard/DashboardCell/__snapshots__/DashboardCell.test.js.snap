// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders dashboard cell 1`] = `
<div
  className="dashboard__cell"
>
  <div
    className="applicant"
  >
    <div
      className="applicant__picture"
    >
      <img
        alt="francis"
        src="https://randomuser.me/api/portraits/med/men/15.jpg"
      />
    </div>
    <div
      className="applicant__data"
    >
      <div>
        First name: 
        francis
      </div>
      <div>
        Last name: 
        stewart
      </div>
      <div>
        Country: 
        bath
      </div>
    </div>
  </div>
  <div
    className="dashboard__cell-controls"
  >
    <div />
    <div />
  </div>
</div>
`;

exports[`Renders dashboard cell with "next" button 1`] = `
<div
  className="dashboard__cell"
>
  <div
    className="applicant"
  >
    <div
      className="applicant__picture"
    >
      <img
        alt="francis"
        src="https://randomuser.me/api/portraits/med/men/15.jpg"
      />
    </div>
    <div
      className="applicant__data"
    >
      <div>
        First name: 
        francis
      </div>
      <div>
        Last name: 
        stewart
      </div>
      <div>
        Country: 
        bath
      </div>
    </div>
  </div>
  <div
    className="dashboard__cell-controls"
  >
    <div />
    <div>
      <button
        id="next"
        onClick={[Function]}
        type="button"
      >
        ›
      </button>
    </div>
  </div>
</div>
`;

exports[`Renders dashboard cell with "previous" button 1`] = `
<div
  className="dashboard__cell"
>
  <div
    className="applicant"
  >
    <div
      className="applicant__picture"
    >
      <img
        alt="francis"
        src="https://randomuser.me/api/portraits/med/men/15.jpg"
      />
    </div>
    <div
      className="applicant__data"
    >
      <div>
        First name: 
        francis
      </div>
      <div>
        Last name: 
        stewart
      </div>
      <div>
        Country: 
        bath
      </div>
    </div>
  </div>
  <div
    className="dashboard__cell-controls"
  >
    <div>
      <button
        id="previous"
        onClick={[Function]}
        type="button"
      >
        ‹
      </button>
    </div>
    <div />
  </div>
</div>
`;

exports[`Renders dashboard cell with both buttons 1`] = `
<div
  className="dashboard__cell"
>
  <div
    className="applicant"
  >
    <div
      className="applicant__picture"
    >
      <img
        alt="francis"
        src="https://randomuser.me/api/portraits/med/men/15.jpg"
      />
    </div>
    <div
      className="applicant__data"
    >
      <div>
        First name: 
        francis
      </div>
      <div>
        Last name: 
        stewart
      </div>
      <div>
        Country: 
        bath
      </div>
    </div>
  </div>
  <div
    className="dashboard__cell-controls"
  >
    <div>
      <button
        id="previous"
        onClick={[Function]}
        type="button"
      >
        ‹
      </button>
    </div>
    <div>
      <button
        id="next"
        onClick={[Function]}
        type="button"
      >
        ›
      </button>
    </div>
  </div>
</div>
`;
